app/main.py

```diff
@@ -1,3 +1,50 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+
+from app.car import Car
+from app.customer import Customer
+from app.shop import Shop
+
+
+def shop_trip() -> None:
+    with open("app/config.json", "r") as config_file:
+        data = json.load(config_file)
+        customers = data["customers"]
+        shops = data["shops"]
+
+    for person in customers:
```

Comment:
So again this main() function is way too big. A better implementation would be like:
1. Make a car as one of the attrs of the customer (would make sense because they are kind of linked in real world)
2. Create shop instances in list comprehension.
3. Move everything below line 32 into a separate method or methods on customer (e.g. find_cheapest_store_and_go_there).
4. Call this method on each newly created instance of a customer.
This way you will make your main function cleaner - and break down logic of code better